editoast/src/views/infra/auto_fixes/speed_section.rs

```diff
@@ -1,19 +1,207 @@
-use super::{new_ref_fix_delete_pair, Fix};
-use crate::schema::{InfraError, InfraErrorType, OSRDObject as _, ObjectRef, SpeedSection};
+use super::{Fix, OrderedOperation};
+use crate::schema::{
+    operation::{CacheOperation, DeleteOperation, Operation, RailjsonObject, UpdateOperation},
+    InfraError, InfraErrorType, OSRDIdentified as _, OSRDObject as _, ObjectRef, SpeedSection,
+};
+use itertools::Itertools as _;
+use json_patch::{Patch, PatchOperation, RemoveOperation};
 use log::debug;
 use std::collections::HashMap;
 
+fn invalid_reference_to_ordered_operation(
+    speed_section: &SpeedSection,
+    object_ref: &ObjectRef,
+) -> Option<OrderedOperation> {
+    let (track_range_idx, _) = speed_section
+        .track_ranges
+        .iter()
+        .enumerate()
+        .find(|(_idx, track_range)| track_range.track.as_str() == object_ref.obj_id)?;
+    Some(OrderedOperation::RemoveTrackRange { track_range_idx })
+}
+
 pub fn fix_speed_section(
     speed_section: &SpeedSection,
     errors: impl Iterator<Item = InfraError>,
 ) -> HashMap<ObjectRef, Fix> {
-    errors
+    let operation = errors
         .filter_map(|infra_error| match infra_error.get_sub_type() {
-            InfraErrorType::EmptyObject => Some(new_ref_fix_delete_pair(speed_section)),
+            InfraErrorType::EmptyObject => Some(OrderedOperation::Delete),
+            InfraErrorType::InvalidReference { reference } => {
+                invalid_reference_to_ordered_operation(speed_section, reference)
+            }
             _ => {
                 debug!("error not (yet) fixable for '{}'", infra_error.get_type());
                 None
             }
         })
+        // Need to invert the ordering because removing from the front would invalidate other indexes
+        .sorted_by_key(|ordered_operation| std::cmp::Reverse(ordered_operation.clone()))
+        .map(|ordered_operation| match ordered_operation {
+            OrderedOperation::RemoveTrackRange { track_range_idx } => {
+                Operation::Update(UpdateOperation {
+                    obj_id: speed_section.get_id().clone(),
+                    obj_type: speed_section.get_type(),
+                    railjson_patch: Patch(vec![PatchOperation::Remove(RemoveOperation {
+                        path: format!("/track_ranges/{track_range_idx}"),
+                    })]),
+                })
+            }
+            OrderedOperation::Delete => {
+                Operation::Delete(DeleteOperation::from(speed_section.get_ref()))
+            }
+        })
+        .map(Some)
+        .reduce(super::reduce_operation)
+        .flatten();
+    operation
+        .and_then(|operation| {
+            let cache_operation = CacheOperation::try_from_operation(
+                &operation,
+                RailjsonObject::SpeedSection {
+                    railjson: speed_section.clone(),
+                },
+            )
+            .ok()?;
```

Comment:
All the interfaces of the `fix_*` functions aren't faillible. That would mean changing all of them. Are we up for that? Is it so important to report errors that would not interrupt the process anyway? Because basically, we wouldn't return a 500 with an error here, we would just ignore this error and handle the other errors. Maybe logging it here would be enough?

Reply:
Right, I forgot about not-fallible `fix_*`.
Then yes, logging would be nice.

Reply:
Added some logging in [3a9480f](https://github.com/osrd-project/osrd/pull/6012/commits/3a9480f90906cd7aa3ca141ab4d1932e293aecb6).